functions.sh

```diff
@@ -311,3 +311,55 @@ upload_updates_image() {
     # Provide the URL of the updates image.
     echo "${httpd_url}updates.img"
 }
+
+export_additional_repo() {
+    # Export additional RPM repository via localhost web server if found in the
+    # /opt/kstest/data/additional_repo well known path. We expect the folder
+    # to just contain a bunch of RPM files and will copy its contents to a tempdir
+    # and then run createrepo on it to make sure all necessary metadata is in place.
+    #
+    # Address of the localhost web server can be sourced from ${tmpdir}/addrepo_url
+    # for use in kernel_args(). Also as this starts a localhost web server do not
+    # forget to shut the server down in cleanup().
+    #
+    # This function expects one argument - path to the tempdir.
+    tmpdir=$1
+
+    # check if additional repo exists on well known path
+
+    if [ -e /opt/kstest/data/additional_repo ]; then
+        # Copy the repo to the tmpdir so
+        # that the createrepo call odes not
+        # mess up the directory on the host.
+        cp -R /opt/kstest/data/additional_repo ${tmpdir}/kstest_additional_repo
```

Comment:
Shouldn't the runner copy or link this directory for each test the same way as the boot.iso? I don't think that the tests should touch anything outside their temporary directory.

Reply:
That's what I'm not sure about as well - I've tried to find out how to properly access the data folder but have not found anything (eq. some env vars, etc.) and using the `/opt/kstest/data/`. Still definitely not ideal even just because we hardcode it in the test code effectively.

As for your suggestion, do you mean changing the runner script to detect the `additional_repo` folder in `data` and do something about it ? Eq. copy it to the `$script` path or something like that ?

I've actually got the impression we don't want to do special casing like that and tests should just fetch stuff from the `data` folder directly. Of course while making sure to use a local copy if they plan to modify the data. But I might have got it wrong. :P